lib/mapObjects/CObjectHandler.cpp

```diff
@@ -457,16 +457,16 @@ IBoatGenerator::EGeneratorState IBoatGenerator::shipyardStatus() const
 		return TILE_BLOCKED; //no available water
 	else if(t->blockingObjects.empty())
 		return GOOD; //OK
-	else if(t->blockingObjects.front()->ID == Obj::BOAT)
+	else if(t->blockingObjects.front()->ID == Obj::TRANSPORT)
 		return BOAT_ALREADY_BUILT; //blocked with boat
 	else
 		return TILE_BLOCKED; //blocked
 }
 
-int IBoatGenerator::getBoatType() const
+TransportId IBoatGenerator::getTransportType() const
```

Comment:
Perhaps make this method pure virtual and reimplement it in CGShipyard?

All other inheritors already use own implementation anyway.
Also - would be great to be able to define what "transport" is built in shipyard in config.

Reply:
this change is only about boats, not about shipyards